Solutions/Corvus.Json.ExtendedTypes/Corvus.Json/Internal/JsonConverter{T}.cs

```diff
@@ -0,0 +1,39 @@
+﻿// <copyright file="JsonConverter{T}.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+using System.Text.Json;
+
+namespace Corvus.Json.Internal;
+
+/// <summary>
+/// Generic json converter for <see cref="IJsonValue{T}"/>.
+/// </summary>
+/// <typeparam name="T">The type of <see cref="IJsonValue{T}"/> to convert.</typeparam>
+public class JsonConverter<T> : System.Text.Json.Serialization.JsonConverter<T>
+    where T : struct, IJsonValue<T>
+{
+    /// <inheritdoc/>
+    public override T Read(
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options)
+    {
+        if (!JsonConverter.EnableInefficientDeserializationSupport)
+        {
+            throw new InvalidOperationException($"Serialization of IJsonValue is not advised. Prefer constructing from a JsonDocument instance to avoid managed allocations. If you require support for integration purposes, enable this with {nameof(JsonConverter.EnableInefficientDeserializationSupport)}.");
```

Comment:
Since serialization works whether the flag is enabled or not, and it's only deserialization that we block by default, this exception should start with the word "Deserialization"

Reply:
Updated.